src/job/job.cpp

```diff
@@ -133,11 +133,18 @@ void job::build_job()
 				limits = std::make_shared<sandbox_limits>(worker_config_->get_limits());
 			}
 
+			// check relativeness of working directory
+			if (!helpers::check_relative(fs::path(sandbox->working_directory))) {
+				throw job_exception("Given working directory for task '" + task_meta->task_id + "' is not relative");
```

Comment:
Better exception message, please. A relative path with ".." will throw this exception as well.